http/server.go

```diff
@@ -186,7 +186,8 @@ func (h *handler) getRand(ctx context.Context, info *chain.Info, round uint64) (
 	h.pendingLk.RUnlock()
 	// If so, prepare, and if we're still sync'd, add ourselves to the list of waiters.
 	if block {
-		ch := make(chan []byte)
+		ch := make(chan []byte, 1)
+		defer close(ch)
```

Comment:
Would it be possible to use the more safe Go pattern where the sender closes the channel, instead of the receiver ? so the receiver can simply `for _ := range myChan { .. }` safely.

Reply:
would you do it with a different channel for the receiver to notify the sender that its context has closed then?